content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -6,12 +6,19 @@ const {GroupedActivityFeed} = require("components/ActivityFeed/ActivityFeed");
 const Spotlight = require("components/Spotlight/Spotlight");
 const Search = require("components/Search/Search");
 const {actions} = require("actions/action-manager");
+let REPORT_RENDER_EVENT = true;
 
 const NewTabPage = React.createClass({
   // TODO: Replace with real search api via addon
   onSearch(value) {
     this.props.dispatch(actions.NotifyPerformSearch(value));
   },
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps.isReady && REPORT_RENDER_EVENT) {
+      this.props.dispatch(actions.NotifyTelemetry("NEWTAB_RENDER"));
```

Comment:
The `componentDidUpdate` function fires every time the component's props or state changes, so it will fire any time the component changes after it is initially ready, such as  when a user deletes an item.